src/hotspot/share/gc/shenandoah/shenandoahEvacTracker.cpp

```diff
@@ -23,44 +23,59 @@
  */
 #include "precompiled.hpp"
 
+#include "gc/shenandoah/shenandoahAgeCensus.hpp"
 #include "gc/shenandoah/shenandoahHeap.inline.hpp"
 #include "gc/shenandoah/shenandoahEvacTracker.hpp"
 #include "gc/shenandoah/shenandoahThreadLocalData.hpp"
 #include "gc/shenandoah/shenandoahRootProcessor.hpp"
 #include "runtime/threadSMR.inline.hpp"
 #include "runtime/thread.hpp"
 
-ShenandoahEvacuationStats::ShenandoahEvacuationStats()
+ShenandoahEvacuationStats::ShenandoahEvacuationStats(bool generational)
   : _evacuations_completed(0), _bytes_completed(0),
     _evacuations_attempted(0), _bytes_attempted(0),
-    _age_table(false) {}
+    _generational(generational) {
+  if (generational && (ShenandoahGenerationalCensusAtEvac || !ShenandoahGenerationalAdaptiveTenuring)) {
```

Comment:
Could we drop `Generational` from these options? It's clear enough (at least to me) that these apply to the generational mode.

Reply:
The code can be called also in non-generational (single generation) mode, and records the same stats. Obviously we want to elide the age census that we do in generational mode in that situation. Hence the option.

Let me know if you had something else in mind here for avoiding the `_generational` attribute or of skinning this cat in a different way.